### PR TITLE
Remove STORAGE_ACCOUNT_KEY setting

### PR DESCRIPTION
**Pull request by mpagot**
Remove any setting usage from terraform.tfvars, and PC libs dedicated to SAP. Removing this small portion of code has no impact as there's no more tests using azure.tfvars at all.

- Related ticket: https://jira.suse.com/browse/TEAM-8710

- Verification run: Change is about a no more used test

### CODE REVIEW

**Review by lilyeyes — APPROVED** [review]
LGTM.

File: data/publiccloud/terraform/sap/azure.tfvars [hunk 1]
```diff
@@ -319,7 +318,7 @@ drbd_image_uri = "https://eisleqasapopenqa.blob.core.windows.net/sle-images/%SLE
 drbd_enabled = false
 
 # Each drbd cluster host IP address (sequential order). If it's not set the addresses will be auto generated from the provided vnet address range
-drbd_ips = ["10.74.1.21", "10.74.1.22"]
+drbd_ips         = ["10.74.1.21", "10.74.1.22"]
```
Comment (m-dati) [inline data/publiccloud/terraform/sap/azure.tfvars:321]: Is this tab needed ?

File: lib/publiccloud/provider.pm [hunk 2]
```diff
@@ -478,7 +478,6 @@ sub terraform_apply {
         my $sap_media = get_required_var('HANA');
         my $sap_regcode = get_required_var('SCC_REGCODE_SLES4SAP');
```
Comment (m-dati) [inline lib/publiccloud/provider.pm:479]: Kind question: given the declared scope in PR header, are parameters like this or `*HANA` too still needed instead?

**Review by Amrysliu — APPROVED** [review]
LGTM

File: data/publiccloud/terraform/sap/azure.tfvars (hunk 1 above)
Comment (BillAnastasiadis) [reply data/publiccloud/terraform/sap/azure.tfvars:321]: I wondered as well, but all other grouped vars in the file (most of which are commented out, but there are a couple uncommented too) use tabs to look uniform, so I think it's fine for consistency

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

File: data/publiccloud/terraform/sap/azure.tfvars (hunk 1 above)
Comment (mpagot) [reply data/publiccloud/terraform/sap/azure.tfvars:321]: It is result of `terraform fmt`. Sorry was not intentional (I have it configured in my editor) but I hope at the same time it is not so annoying.

File: lib/publiccloud/provider.pm (hunk 2 above)
Comment (mpagot) [reply lib/publiccloud/provider.pm:479]: I hope to be able to remove more code from this file in the near future, But it is beyond the purpose of the associated ticket.

**Pull request merged**